File: backend-spring-boot/src/main/java/no/vicx/backend/calculator/RemoveOldEntriesTask.java
package no.vicx.backend.calculator;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
public record RemoveOldEntriesTask(CalculatorService calculatorService) {
    private static final Logger LOG = LoggerFactory.getLogger(RemoveOldEntriesTask.class);

    @Scheduled(fixedRateString = "${app.calculator.rate}")
    public void removeOldCalculatorEntries() {
        LOG.info("Removing old calculator entries");
        calculatorService.deleteOldAnonymousCalculations();
    }
}

File: backend-spring-boot/src/main/kotlin/no/vicx/backend/calculator/RemoveOldEntriesTask.kt
package no.vicx.backend.calculator

import no.vicx.backend.loggerFor
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component


@Component
class RemoveOldEntriesTask(
    private val calculatorService: CalculatorService
) {
    @Scheduled(fixedRateString = "\${app.calculator.rate}")
    fun removeOldCalculatorEntries() {
        log.info("Removing old calculator entries")
        calculatorService.deleteOldAnonymousCalculations()
    }

    companion object {
        private val log = loggerFor<RemoveOldEntriesTask>()
    }
}
